### PR TITLE
Update the schema to match `metadata`

### PR DESCRIPTION
**Pull request by vmarkovtsev**

### CODE REVIEW

**Comment by dpordomingo** [comment]
I pulled this PR, built and try,
```
$ make docker-build;
$ make run-api
```
and got: https://sentry.io/organizations/athenianco/issues/1447689123/?project=1867351&query=is%3Aunresolved
I'm not sure if it will help you.

**Comment by dpordomingo** [comment]
I pulled this PR, built and try,
```
$ make docker-build;
$ make run-api

make run-api
docker run --rm -e DB_DIR=/io -v/data/src/athenian/athenian-api/server/tests:/io --env-file .env --entrypoint python3 athenian/api:dev /server/tests/gen_sqlite_db.py
docker run --rm -p 8080:8080 -v/data/src/athenian/athenian-api/server/tests:/io --env-file .env athenian/api:dev --ui --metadata-db=sqlite:///io/mdb.sqlite --state-db=sqlite:///io/sdb.sqlite
INFO:athenian.api:['/server/athenian/api/__main__.py', '--ui', '--metadata-db=sqlite:///io/mdb.sqlite', '--state-db=sqlite:///io/sdb.sqlite']
INFO:athenian.api:Version 0.0.1
INFO:alembic.runtime.migration:Context impl SQLiteImpl.
INFO:alembic.runtime.migration:Will assume non-transactional DDL.
Traceback (most recent call last):
  File "/usr/lib/python3.7/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/usr/lib/python3.7/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/server/athenian/api/__main__.py", line 6, in <module>
    main()
  File "/server/athenian/api/__init__.py", line 206, in main
    check_schema_version(args.state_db)
  File "/server/athenian/api/models/state/__init__.py", line 27, in check_schema_version
    raise StateDBSchemaVersionMismatchError("Required: %s Connected: %s" % (req_rev, real_rev))
athenian.api.models.state.StateDBSchemaVersionMismatchError: Required: a8d2bdb184f0 Connected: None
```
and got: https://sentry.io/organizations/athenianco/issues/1447689123/?project=1867351&query=is%3Aunresolved

The `Required: a8d2bdb184f0 Connected: None` message, seems to be related to invitations creation [`a8d2bdb184f0_create_invitations_table.py#L13`](https://github.com/athenianco/athenian-api/blob/master/server/athenian/api/models/state/versions/a8d2bdb184f0_create_invitations_table.py#L13)

I'm not sure if it will help you.

**Comment by vmarkovtsev** [comment]
Yeah, I hit this, too. There is a new problem after fixing the problem :) I will backport the fix to resolve ENG-156.

File: server/athenian/api/models/metadata/github/__init__.py [hunk 1]
```diff
@@ -1,23 +1,31 @@
...
+
 class BodyMixin:
     body = Column("body", Text)
 
 
 class UpdatedMixin:
-    created_at = Column("created_at", DateTime(True))
-    updated_at = Column("updated_at", DateTime(True))
+    created_at = Column("created_at", TIMESTAMP(True))
+    updated_at = Column("updated_at", TIMESTAMP(True))
```
Comment (se7entyse7en) [inline server/athenian/api/models/metadata/github/__init__.py]: I thought that in the end we were going to remove tz, doesn't we?

File: server/athenian/api/models/metadata/github/__init__.py [hunk 2]
```diff
@@ -1,23 +1,31 @@
...
+
+class IDMixin:
+    id = Column("id", BigInteger, primary_key=True)
     node_id = Column("node_id", Text)
 
 
+class DeliveryMixin:
+    delivery_id = Column("delivery_id", Text, nullable=False)
+    action = Column("action", Text)
+    timestamp = Column("timestamp", TIMESTAMP(True))
```
Comment (se7entyse7en) [inline server/athenian/api/models/metadata/github/__init__.py]: Same comment wrt to tz.

File: server/athenian/api/models/metadata/github/__init__.py [hunk 3]
```diff
@@ -1,23 +1,31 @@
-from sqlalchemy import ARRAY, BigInteger, Boolean, Column, DateTime, Integer, String, Text
+from sqlalchemy import ARRAY, BigInteger, Boolean, Column, DateTime, ForeignKey, Integer, Text, \
+    TIMESTAMP
```
Comment (se7entyse7en) [inline server/athenian/api/models/metadata/github/__init__.py]: Isn't it possible to have only `DateTime`s or only `TIMESTAMP`?

File: server/athenian/api/controllers/reposet_controller.py [hunk 4]
```diff
@@ -97,6 +98,22 @@
         return e.response
     rss = await request.sdb.fetch_all(
         select([RepositorySet]).where(RepositorySet.owner == id))
+    if len(rss) == 0:
+        # new account, discover the repos the metadata DB
+        async with request.mdb.connection() as conn:
+            iid = await conn.fetch_val(
+                select([InstallationOwner.install_id])
+                .where(InstallationOwner.user_id == "user_id_from_Auth0"))
```
Comment (se7entyse7en) [inline server/athenian/api/controllers/reposet_controller.py]: Isn't the id taken from the `request`?

File: server/athenian/api/controllers/miners/github/pull_request.py [hunk 5]
```diff
@@ -35,33 +37,55 @@ def __init__(self, prs: pd.DataFrame, reviews: pd.DataFrame, review_comments: pd
...
+                         PullRequestReview.repository_fullname.in_(repositories))),
+                conn, PullRequestReview)
+            review_comments = await read_sql_query(select([PullRequestComment]).where(
+                sql.and_(PullRequestComment.pull_request_number.in_(numbers),
+                         PullRequestComment.repository_fullname.in_(repositories))),
+                conn, PullRequestComment)
+            commits = await read_sql_query(select([PullRequestCommit]).where(
+                sql.and_(PullRequestCommit.pull_request_number.in_(numbers),
+                         PullRequestCommit.repository_fullname.in_(repositories))),
+                conn, PullRequestCommit)
```
Comment (se7entyse7en) [inline server/athenian/api/controllers/miners/github/pull_request.py]: I'd add a
```
def _get_filtered_data(model, pr_numbers, repos):
    return await read_sql_query(select([model]).where(
                sql.and_(model.pull_request_number.in_(pr_numbers),
                         model.repository_fullname.in_(repos))),
                conn, model)
```

File: server/athenian/api/controllers/miners/github/pull_request.py [hunk 6]
```diff
@@ -35,33 +37,55 @@ def __init__(self, prs: pd.DataFrame, reviews: pd.DataFrame, review_comments: pd
...
+        return cls(prs, reviews, review_comments, commits, time_to)
+
+    def __iter__(self) -> Generator[Tuple[pd.Series, pd.DataFrame, pd.DataFrame, pd.DataFrame],
+                                    None, None]:
         """Iterate over the information collected for individual pull requests."""
+        r_repo_key = PullRequest.repository_fullname.key
```
Comment (se7entyse7en) [inline server/athenian/api/controllers/miners/github/pull_request.py]: I think this should be:
```suggestion
        r_repo_key = PullRequestReview.repository_fullname.key
```

File: server/athenian/api/controllers/miners/github/pull_request.py [hunk 7]
```diff
@@ -35,33 +37,55 @@ def __init__(self, prs: pd.DataFrame, reviews: pd.DataFrame, review_comments: pd
...
+                & (self._reviews[r_repo_key] == pr_repo)]  # noqa: W503
             review_comments = self._review_comments[
-                self._review_comments["pull_request_number"] == pr["number"]]
-            yield pr, reviews, review_comments
+                (self._review_comments[ce_pull_request_number_key] == pr_number)
+                & (self._review_comments[ce_repo_key] == pr_repo)]  # noqa: W503
+            commits = self._commits[
+                (self._commits[ci_pull_request_number_key] == pr_number)
+                & (self._commits[ci_repo_key] == pr_repo)]  # noqa: W503
+            yield pr, reviews, review_comments, commits
```
Comment (se7entyse7en) [inline server/athenian/api/controllers/miners/github/pull_request.py]: I think that something as follows would be more readable (btw I didn't test it):
```
    def __iter__(self) -> Generator[Tuple[pd.Series, pd.DataFrame, pd.DataFrame, pd.DataFrame],
                                    None, None]:
        keys = {
            'reviews': {
                'pr': PullRequestReview.pull_request_number.key,
                'repo': PullRequestReview.repository_fullname.key,
            },
            'review_comments': {
                'pr': PullRequestComment.pull_request_number.key,
                'repo': PullRequestComment.repository_fullname.key,
            },
            'commits': {
                'pr': PullRequestCommit.pull_request_number.key,
                'repo': PullRequestCommit.repository_fullname.key 
            }
        }

        def _get_filtered(what, pr_number, pr_repo):
            df = getattr(self, f'_{what}')
            pr_key, repo_key = keys[what]['pr'], keys[what]['key']
            return df[(df[pr_key] == pr_number) & (df[repo_key] == pr_repo)]
        
        number_key = PullRequest.number.key
        repo_key = PullRequest.repository_fullname.key
        for _, pr in self._prs.iterrows():
            pr_number, pr_repo = pr[number_key], pr[repo_key]

            reviews = _get_filtered('reviews', pr_number, pr_repo)
            reviews_comments = _get_filtered('review_comments', pr_number, pr_repo)
            commits = _get_filtered('commits', pr_number, pr_repo)

            yield pr, reviews, review_comments, commits
```

File: server/athenian/api/controllers/reposet_controller.py (hunk 4 above)
Comment (vmarkovtsev) [reply server/athenian/api/controllers/reposet_controller.py]: This is exactly the place where I am blocked: this requires knowing the exact format of the JWT token which, from @dennwc's words, contains the GitHub ID.

There is a (crappy) workaround to query the mgmt API - it returns the nice `identities` list.

File: server/athenian/api/controllers/miners/github/pull_request.py (hunk 5 above)
Comment (vmarkovtsev) [reply server/athenian/api/controllers/miners/github/pull_request.py]: A second pair of eyes rocks!

File: server/athenian/api/controllers/reposet_controller.py (hunk 4 above)
Comment (se7entyse7en) [reply server/athenian/api/controllers/reposet_controller.py]: okay, sorry for dealying this, asap I'll ping you when integrating webapp <> api.

**Comment by vmarkovtsev** [comment]
@lwsanty Merging this as soon as the infra is ready.

**Pull request merged**